src/ducks/Signals/signalFormManager/signalCrudForm/formParts/TriggerMetricTypesResolver.js

```diff
@@ -0,0 +1,53 @@
+import React from 'react'
+import TriggerFormAssetWallet from './TriggerFormAssetWallet'
+import TriggerFormTrendingWordsTypes from './TriggerFormTrendingWordsTypes'
+import { ETH_WALLET, TRENDING_WORDS } from '../../../utils/constants'
+import TriggerFormHistoricalBalance from './TriggerFormHistoricalBalance'
+
+const TriggerMetricTypesResolver = ({
+  address,
+  values,
+  metaFormSettings,
+  setFieldValue
+}) => {
+  const { metric, target } = values
+
+  let TypeComponent
+
+  let checkPossibleTarget = () => {
+    if (Array.isArray(target)) {
+      setFieldValue(
+        'target',
+        target.length > 0 ? target[0] : metaFormSettings.target.value
+      )
+    }
+  }
+
+  switch (metric.value) {
+    case TRENDING_WORDS: {
+      TypeComponent = TriggerFormTrendingWordsTypes
+      break
+    }
+    case ETH_WALLET: {
+      checkPossibleTarget()
+      TypeComponent = TriggerFormHistoricalBalance
+      break
+    }
+    default: {
+      checkPossibleTarget()
+      TypeComponent = TriggerFormAssetWallet
+      break
+    }
+  }
```

Comment:
1. `TypeComponent` could be initialised via dictionary;
2. Instead of creating `checkPossibleTarget` closure and then calling it, simple `if (metric.value !== TRENDING_WORDS)` statement could be placed before `TypeComponent` initialisation.